repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -4108,6 +4510,224 @@ public void updateClassifications(EntityMutationContext context, String guid, Li
+               |-------------|-------------| */
+
+            Boolean currentTagPropagation = currentClassification.isPropagate();
+            Boolean updatedTagPropagation = classification.isPropagate();
+            Boolean currentRestrictPropagationThroughLineage = currentClassification.getRestrictPropagationThroughLineage();
+            Boolean updatedRestrictPropagationThroughLineage = classification.getRestrictPropagationThroughLineage();
+            Boolean currentRestrictPropagationThroughHierarchy = currentClassification.getRestrictPropagationThroughHierarchy();
+            Boolean updatedRestrictPropagationThroughHierarchy = classification.getRestrictPropagationThroughHierarchy();
+            if (updatedRestrictPropagationThroughLineage == null) {
+                updatedRestrictPropagationThroughLineage = currentRestrictPropagationThroughLineage;
+                classification.setRestrictPropagationThroughLineage(updatedRestrictPropagationThroughLineage);
+            }
+            if (updatedRestrictPropagationThroughHierarchy == null) {
+                updatedRestrictPropagationThroughHierarchy = currentRestrictPropagationThroughHierarchy;
+                classification.setRestrictPropagationThroughHierarchy(updatedRestrictPropagationThroughHierarchy);
+            }
+
+            if (updatedTagPropagation)
+                entityRetriever.validatePropagationRestrictionOptions(updatedRestrictPropagationThroughLineage, updatedRestrictPropagationThroughHierarchy);
+
+            if ((!Objects.equals(updatedRemovePropagations, currentRemovePropagations) ||
+                    !Objects.equals(currentTagPropagation, updatedTagPropagation) ||
+                    !Objects.equals(currentRestrictPropagationThroughLineage, updatedRestrictPropagationThroughLineage) ||
+                    !Objects.equals(currentRestrictPropagationThroughHierarchy, updatedRestrictPropagationThroughHierarchy)) &&
+                    taskManagement != null && DEFERRED_ACTION_ENABLED) {
+
+                propagationType = CLASSIFICATION_PROPAGATION_ADD;
+                if(currentRestrictPropagationThroughLineage != updatedRestrictPropagationThroughLineage || currentRestrictPropagationThroughHierarchy != updatedRestrictPropagationThroughHierarchy){
+                    propagationType = CLASSIFICATION_REFRESH_PROPAGATION;
+                }
+                if (removePropagation || !updatedTagPropagation) {
+                    propagationType = CLASSIFICATION_PROPAGATION_DELETE;
+                }
+
+                String  currentUser = RequestContext.getCurrentUser();
+                String  entityGuid  = GraphHelper.getGuid(entityVertex);
+
+                Map<String, Object> taskParams  = new HashMap<String, Object>() {{
+                    put(PARAM_ENTITY_GUID, entityGuid);
+                }};
+
+                taskManagement.createTaskV2(propagationType, currentUser, taskParams, classification.getTypeName(), entityGuid);
```

Comment:
We are not passing currentRestrictPropagationThroughLineage & updatedRestrictPropagationThroughHierarchy for task, can someone please test the propagationType = CLASSIFICATION_PROPAGATION_ADD; case form this flow & see if add tag propagation task does the job as expected?